### ISSUE TITLE
Bug: Can't configure project service for files with custom extensions

### ISSUE THREAD

**Issue by tjzel** [typescript-eslint/typescript-eslint#11707]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug) and found none that matched my issue.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Environment

Node version: v22.14.0
npm version: v10.9.2
Local ESLint version: v9.37.0 (Currently used)
Global ESLint version: Not found
Operating System: darwin 24.6.0

### What did you do?

I'm trying to configure ESLint in a repository which has platform-specific versions of files, i.e. there is:

- `Foo.tsx`
- `Foo.web.tsx`
- `Bar.tsx`
- `Bar.native.tsx`

However, I can't configure ESLint to properly lint them and I'm stuck on `Parsing error: /Users/user/eslint-repro/src/Foo.web.tsx was not found by the project service. Consider either including it in the tsconfig.json or including it in allowDefaultProject`.

The key aspect here is that I have several `tsconfig.json` files, respective to each platform.

<details><summary>Simplified TS configs look like this</summary>
<p>

```ts
// tsconfig.native.json
{
  "compilerOptions": {
    "moduleSuffixes": [".native", ".ios", ".android", ""] // disallow .web files
  },
  "include": ["src"],
  "exclude": ["src/**/*.web.ts", "src/**/*.web.tsx"]
}

// tsconfig.web.json
{
  "compilerOptions": {
    "moduleSuffixes": [".web", ""] // disallow .native, .ios, .android files
  },
  "include": ["src"],
  "exclude": [
    "src/**/*.native.ts",
    "src/**/*.native.tsx",
    "src/**/*.ios.ts",
    "src/**/*.ios.tsx",
    "src/**/*.android.ts",
    "src/**/*.android.tsx"
  ]
}

// tsconfig.json (default for IDE features)
{
  "extends": "./tsconfig.native.json"
}
```

</p>
</details> 

<details><summary>This is my base config that doesn't work</summary>
<p>

```ts
import tsEslint from "typescript-eslint";
import { defineConfig } from "eslint/config";
import importPlugin from "eslint-plugin-import";

/** @type {import('typescript-eslint').ConfigWithExtends[]} */
export default defineConfig(
  tsEslint.configs.recommendedTypeChecked,
  importPlugin.flatConfigs.recommended,
  {
    languageOptions: {
      parserOptions: {
        projectService: true,
      },
    },
    settings: {
      "import/resolver": {
        typescript: {
          alwaysTryTypes: true,
        },
      },
    },
  }
);
```

</p>
</details>

I've tried various options like explicit ignores, toying with `projectService.defaultProject` but nothing seems to work.

<details><summary>More fine grained config</summary>
<p>

```ts
import tsEslint from "typescript-eslint";
import { defineConfig } from "eslint/config";
import importPlugin from "eslint-plugin-import";

/** @type {import('typescript-eslint').ConfigWithExtends[]} */
export default defineConfig(
  {
    languageOptions: {
      parserOptions: {
        projectService: true,
      },
    },
    settings: {
      "import/resolver": {
        typescript: {
          alwaysTryTypes: true,
        },
      },
    },
    extends: [
      tsEslint.configs.recommendedTypeChecked,
      importPlugin.flatConfigs.recommended,
    ],
    files: ["**/*.ts", "**/*.tsx"],
    ignores: ["**/*.web.ts", "**/*.web.tsx"],
  },
  {
    languageOptions: {
      parserOptions: {
        projectService: {
          defaultProject: "tsconfig.web.json",
        },
      },
    },
    settings: {
      "import/resolver": {
        typescript: {
          alwaysTryTypes: true,
          project: "tsconfig.web.json",
        },
      },
    },
    extends: [
      tsEslint.configs.recommendedTypeChecked,
      importPlugin.flatConfigs.recommended,
    ],
    files: ["**/*.web.ts", "**/*.web.tsx"],
    ignores: ["**/*.native.ts", "**/*.native.tsx"],
  }
);
```

</p>
</details> 

### What did you expect to happen?

All files:

- `Foo.tsx`
- `Foo.web.tsx`
- `Bar.tsx`
- `Bar.native.tsx`

are linted when I invoke `yarn eslint src`

### What actually happened?

> Parsing error: /Users/user/eslint-repro/src/Foo.web.tsx was not found by the project service. Consider either including it in the tsconfig.json or including it in allowDefaultProject.

### Reproduction Repository Link

https://github.com/tjzel/eslint-parser-repro

### Repro Steps

1) clone the repo
2) `yarn install`
3) `yarn eslint src`

### Versions

| package                                | version |
| -------------------------------------- | ------- |
| `typescript-eslint`     | `8.46.1` |
| `TypeScript`                           | `5.9.3` |
| `ESLint`                               | `9.37.0` |
| `node`                                 | `22.14.0` |

**Comment by bradzacher** [comment]
You should NOT be using the "default project" for this usecase. That option is an escape hatch intended for files that exist outside your regular code - for example your eslint config file.

It is NOT intended as a way to include multiple, custom named tsconfig files.

---

The best way to make this work would be to set your root `tsconfig.json` so that it includes all of your files.

Or if you simply must stick with your current setup, then you can instead use `project` instead of `projectService` and pass the tsconfig paths in.

**Comment by tjzel** [comment]
Hi @bradzacher, thanks for replying. That's an interesting note about `defaultProject`, thanks for explaining.

I misunderstood the docs and didn't realize that `projectService` is mutually exclusive with `project`. Using just `project` works exactly as expected, thanks for help!
